data/food/export_agb/export_builder.py

```diff
@@ -70,7 +77,8 @@ def search(name):
             "id": activity["id"],
             "name": activity["name"],
             "categories": [c for c in activity["categories"] if c != "ingredient"],
-            "default": search(activity["search"])["Process identifier"],
+            "complex": True if "ratio" in activity.keys() else False,
```

Comment:
Also if that's really what you want you can simply write `"complex": "ratio" in activity.keys()`. 

The thing I'm struggling understanding the most is where this JSON property is used? Hint?

Reply:
I've just read in the description

> added a complex tag to ingredients to explicit if it's complex or not. (Arguable)

So maybe what we want is a proper JSON data structure to declare the type of ingredient? If so — and that'd be a larger refactoring — we could go with something like this for simple ingredients:

```json
{
  "id": "onion",
  "name": "Oignon",
  …
  "construct": {
    "type": "simple"
  }
}
```

and for complex ones:

```json
{
  "id": "butter-organic",
  "name": "Beurre bio",
  …
  "construct": {
    "type": "complex",
    "default": "milk",
    "organic": "milk-organic",
    "ratio": 6.58
  }
}
```

Thoughts @cccomb @magopian?

Reply:
Another idea is to add an ingredient category called "complex ingredient".
Otherwise I've not yet understood why we need a "complex" attribute. It introduces the possibility to have an inconsistency by defining complex=True on an ingredient without ratio or the opposite

Reply:
Yeah the `complex` attribute should definitely go away.

As for the `ingredient-complex` new category, it smells a little complicated vs. a simple `construct` data structure explicitly expressing the different required configuration elements; decoding will be bullet-proof and easy to map to proper Elm types at least, if that's ever a concern.

Reply:
agreed, I remove the `complex` attribute